pkg/api/norman/customization/aks/handler.go

```diff
@@ -0,0 +1,232 @@
+			return
+		}
+	} else if req.Method == http.MethodPost {
+		if errCode, err := h.getCredentialsFromBody(req, capa); err != nil {
+			handleErr(writer, errCode, err)
+			return
+		}
+	} else {
+		handleErr(writer, http.StatusBadRequest, fmt.Errorf("cannot access Azure API without credentials to authenticate"))
+		return
+	}
+
+	var serialized []byte
+	var errCode int
+	var err error
+
+	resourceType := mux.Vars(req)["resource"]
+
+	switch resourceType {
+	case "aksVersions":
+		if serialized, errCode, err = listKubernetesVersions(req.Context(), capa); err != nil {
+			logrus.Debugf("[aks-handler] error getting kubernetes versions: %v", err)
+			handleErr(writer, errCode, err)
+			return
+		}
+		writer.Write(serialized)
+	case "aksVirtualNetworks":
+		if serialized, errCode, err = listVirtualNetworks(req.Context(), capa); err != nil {
+			logrus.Debugf("[aks-handler] error getting networks: %v", err)
+			handleErr(writer, errCode, err)
+			return
+		}
+		writer.Write(serialized)
+	default:
+		handleErr(writer, httperror.NotFound.Status, fmt.Errorf("invalid endpoint %v", resourceType))
+	}
+}
+
+func (h *handler) getCloudCredential(req *http.Request, cap *Capabilities, credID string) (int, error) {
+	ns, name := ref.Parse(credID)
+	if ns == "" || name == "" {
+		logrus.Debugf("[AKS] invalid cloud credential ID %s", credID)
+		return http.StatusBadRequest, fmt.Errorf("invalid cloud credential ID %s", credID)
+	}
+
+	var accessCred client.CloudCredential //var to check access
+	if err := access.ByID(h.generateAPIContext(req), &schema.Version, client.CloudCredentialType, credID, &accessCred); err != nil {
+		apiError, ok := err.(*httperror.APIError)
+		if !ok {
+			return httperror.NotFound.Status, err
+		}
+		if apiError.Code.Status == httperror.NotFound.Status {
+			return httperror.InvalidBodyContent.Status, fmt.Errorf("cloud credential not found")
+		}
+		if apiError.Code.Status != httperror.PermissionDenied.Status {
+			return httperror.InvalidBodyContent.Status, err
+		}
+		var clusterID string
+		if clusterID = req.URL.Query().Get("clusterID"); clusterID == "" {
+			return httperror.InvalidBodyContent.Status, fmt.Errorf("cloud credential not found")
+		}
+		if errCode, err := h.clusterCheck(h.generateAPIContext(req), clusterID, credID); err != nil {
+			return errCode, err
+		}
+	}
+
+	cc, err := h.secretsLister.Get(ns, name)
+	if err != nil {
+		logrus.Debugf("[AKS] error accessing cloud credential %s", credID)
+		return httperror.InvalidBodyContent.Status, fmt.Errorf("error accessing cloud credential %s", credID)
+	}
+	cap.SubscriptionID = string(cc.Data["azurecredentialConfig-subscriptionId"])
+	cap.ClientID = string(cc.Data["azurecredentialConfig-clientId"])
+	cap.ClientSecret = string(cc.Data["azurecredentialConfig-clientSecret"])
+
+	cap.TenantID = req.URL.Query().Get("tenantId")
+	if cap.TenantID == "" {
+		logrus.Debugf("[AKS] error getting tenantId")
+		return http.StatusBadRequest, fmt.Errorf("error getting tenantId")
```

Comment:
I thought we stopped using tenantID ? The cloud credential won't have one.

Reply:
Tenant ID is needed, it is part of AKSConfig